Add a note and diagram to clarify legacy mouse pointer behavior.

mustaqahmed opened the pull request:
This shows the effective position of legacy mouse pointer when two primary pointers are active at the same time, and points out the disconnection between `pointerout` and `mouseout` events.

Closes #454.


<!--
    This comment and the below content is programmatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/mustaqahmed/pointerevents/pull/458.html" title="Last updated on Oct 26, 2022, 7:34 PM UTC (5f4cb35)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/pointerevents/458/1074161...mustaqahmed:5f4cb35.html" title="Last updated on Oct 26, 2022, 7:34 PM UTC (5f4cb35)">Diff</a>

mustaqahmed committed:
Add a note and diagram to clarify legacy mouse pointer behavior.

This shows the effective position of legacy mouse pointer when two
primary pointers are active at the same time, and points out the
disconnection between `pointerout` and `mouseout` events.

mustaqahmed commented:
@flackr and @patrickhlauke PTAL.  I will fine-tune on the animated GIF once the text seems okay.

mustaqahmed committed:
Clarify effective position of legacy mouse.

mustaqahmed committed:
Update legacy pointer animation video.

mustaqahmed committed:
Switch to webm video from gif.

flackr commented on index.html:
```diff
@@ -1204,6 +1204,32 @@ <h2><dfn>Tracking the effective position of the legacy mouse pointer</dfn></h2>
+                <img src="images/legacy-pointer-animation.webm"
+                     alt="legacy mouse pointer animation">
+                <figcaption>
+                  Simultaneous mouse and touch pointers causing legacy mouse to move between the two pointers.
```
This could really use an explanation of the orange pointer (being the simulated pointer) and the white cursor and pointer.

flackr replied:
Actually I suppose this is mentioned above, but it wouldn't hurt to clarify, e.g.

 Simultaneous mouse (white cursor) and touch pointer (white pointer) causing legacy mouse (orange cursor) to move between the two pointers.

mustaqahmed replied:
Done.

mustaqahmed committed:
Caption update.

patrickhlauke committed:
Reorganise section

Instead of repeating the same concepts in slightly different way in the note, replace the possibly a bit obscure original wording with some of the clearer prose.
Use `<video>` element for the `.webm` rather than `<img>` (as, at least for me, this didn't appear to be supported in the browser)

patrickhlauke commented:
Did some tweaking/rewording

mustaqahmed commented on index.html:
```diff
@@ -1196,14 +1196,33 @@ <h1><dfn data-lt="compatibility mouse events">Compatibility mapping with mouse e
                 <li>Set <a data-lt="effective position of the legacy mouse pointer">effective legacy mouse pointer position</a> to |T|.</li>
             </ol>
+            <div class="note">
+              <p>The following animation illustrates how a user agent can dispatch compatibility mouse events that reconcile two active primary pointers with an idealised legacy mouse input.</p>
```
I think we need to highlight why the spec deliberately leaves out a direct mapping of legacy transition events (over/enter/out/leave events).  The last line in the spec issue reflects a similar confusion we saw before (perhaps internally in Chrome, can't recall).

I am working on a fix now.

mustaqahmed replied:
Done, please take a look.

patrickhlauke replied:
Thanks...these changes look great to me

mustaqahmed committed:
Explain no direct mapping for transition events

patrickhlauke commented on index.html:
```diff
@@ -1196,7 +1196,7 @@ <h1><dfn data-lt="compatibility mouse events">Compatibility mapping with mouse e
         <section>
             <h2><dfn>Tracking the effective position of the legacy mouse pointer</dfn></h2>
-            <p>While only <a>primary pointers</a> can produce compatibility mouse events, <a href="#multiple-primary-pointers">multiple primary pointers</a> can be active simultaneously, each producing its own compatibility mouse events. For compatibility with scripts relying on MouseEvents, mouse transition events (<code>pointerover</code>, <code>pointerout</code>, <code>pointerenter</code> and  ...
+            <p>While only <a>primary pointers</a> can produce compatibility mouse events, <a href="#multiple-primary-pointers">multiple primary pointers</a> can be active simultaneously, each producing its own compatibility mouse events. For compatibility with scripts relying on MouseEvents, the mouse transition events (<code>mouseover</code>, <code>mouseout</code>, <code>mouseenter</code> and <c ...
```
Good catch listing the actual `mouse*` events here, rather than `pointer*` ones

patrickhlauke commented on index.html:
```diff
@@ -1205,23 +1205,28 @@ <h2><dfn>Tracking the effective position of the legacy mouse pointer</dfn></h2>
+                and <code>pointerleave</code>) to corresponding legacy mouse transition events (i.e., <code>mouseover</code>,
+                <code>mouseout</code>, <code>mouseenter</code> and <code>mouseleave</code>).  The following animation illustrates a case
+                where a user agent needs to dispatch more legacy mouse transition events than pointer transition events to be able to
+                reconcile two primary pointers using a single legacy mouse input.</p>
```
Looking good, thanks
